src/Symfony/Bridge/Doctrine/Types/DatePointType.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Types;
+
+use DateTimeInterface;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\DateTimeImmutableType;
+use Symfony\Component\Clock\DatePoint;
+
+final class DatePointType extends DateTimeImmutableType
+{
+    public const NAME = 'date_point';
+
+    public function convertToPHPValue(mixed $value, AbstractPlatform $platform): ?DatePoint
+    {
+        if (null === $value || $value instanceof DatePoint) {
+            return $value;
+        }
+
+        $value = parent::convertToPHPValue($value, $platform);
+        if ($value instanceof DateTimeInterface) {
+            return DatePoint::createFromInterface($value);
+        }
+
+        throw new ConversionException();
```

Comment:
```suggestion
        $value = parent::convertToPHPValue($value, $platform);
        if (null === $value) {
            return null;
        }

        return DatePoint::createFromInterface($value);
```

The upstream type already performs all the prechecks that we need and we can be certain that it returns either null or a `DateTimeImmutable` instance. This allows us to simplify the implementation a bit and get rid of the dead exception.